### ISSUE TITLE
Stylelint: Disallow "union" sass selectors

### ISSUE THREAD

**Issue by ryelle** [WordPress/wporg-repo-tools#25]
Should we add the [`selector-no-union-class-name`](https://github.com/stylelint-scss/stylelint-scss/blob/master/src/rules/selector-no-union-class-name/README.md) rule to disallow inline `&` in CSS selectors? This would prevent incomplete selectors in SCSS files, but still allow combining selectors.

For example:

```scss
.class {
    &-union { /* not allowed */ }
    &.child-class { /* this is okay */ }
}
```

**Comment by adamwoodnz** [comment]
I agree this can be problematic in large blocks where one can lose sight of the prefix, and there is no way to define what is the right size of block for it to be ok. Having a rule in the tooling means it's one less thing to think about when reviewing so I'm for it 👍

**Comment by iandunn** [comment]
💯 , I find those a much harder to search for, and even when just reading it introduces a tiny bit of friction. I don't really see any significant benefit to them.
